### PR TITLE
collex to case pubsub communication changes

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?

# How to test?

# Trello
https://trello.com/c/LfRNodWI/528-s35a-replace-rabbit-with-pub-sub-collection-exercise-to-case

### CODE REVIEW

File: _infra/helm/case/templates/deployment.yaml [hunk 1]
```diff
@@ -127,10 +127,6 @@ spec:
             value: /var/secrets/google/credentials.json
           - name: GOOGLE_CLOUD_PROJECT
             value: "{{ .Values.gcp.project }}"
-          - name: GCP_PROJECT
```
Comment (insacuri) [inline _infra/helm/case/templates/deployment.yaml:130]: If you delete this then it needs to be replaced with something.  Currently the project field is populated by 
```
gcp:
  project:
```
I know theres a google_cloud_project field but it's not being used by the pubsub stuff

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java [hunk 2]
```diff
@@ -1,30 +1,106 @@
...
+  @Autowired private ObjectMapper objectMapper;
   @Autowired private CaseService caseService;
+  @Autowired AppConfig appConfig;
+
+  /**
+   * To process SampleUnitParents read from pubsub This creates application ready event listener to
+   * provide a active subscription for the new case creation against the receiving SampleUnitParents
+   */
+  @EventListener(ApplicationReadyEvent.class)
+  public void acceptSampleUnit() throws IOException {
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:37]: Minor, but maybe this should be something like `createCaseCreationSubscription()` or something that better describes what it's doing?

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java [hunk 3]
```diff
@@ -1,30 +1,106 @@
...
+    log.debug("received CaseCreation Message from pubsub");
+    // Instantiate an asynchronous message receiver.
+    MessageReceiver receiver =
+        (PubsubMessage message, AckReplyConsumer consumer) -> {
+          // Handle incoming message, then ack the received message.
+          log.with(message.getMessageId()).info("Receiving message ID from PubSub");
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:43]: Is the ID actually useful?  If so, should it be at info level?

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java [hunk 4]
```diff
@@ -1,30 +1,106 @@
...
+          log.with(message.getData().toString()).debug("Receiving data from PubSub ");
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:44]: Remove extra space after PubSub

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java [hunk 5]
```diff
@@ -1,30 +1,106 @@
...
    */
-  @Transactional
-  @ServiceActivator(inputChannel = "caseTransformed", adviceChain = "caseRetryAdvice")
-  public void acceptSampleUnit(SampleUnitParent caseCreation) {
-    log.debug("received CaseCreation Message from queue");
-    caseService.createInitialCase(caseCreation);
+  private ProjectSubscriptionName getCaseCreationSubscriptionName() {
+    String project = appConfig.getGcp().getProject();
+    String subscriptionId = appConfig.getGcp().getCaseNotificationSubscription();
+    log.info(
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:99]: Is this worth it?  It logs out basically the same information once it's finished (the line with `log.with(subscriber.getSubscriptionNameString())`)

**Review by insacuri — CHANGES_REQUESTED** [review]
Mostly looks good apart from a few things.

File: _infra/helm/case/templates/deployment.yaml (hunk 1 above)
Comment (creativeappdevamit) [reply _infra/helm/case/templates/deployment.yaml:130]: no this was a duplicate. you created this key down in the deployment.yaml I missed to see it.

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java (hunk 2 above)
Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:37]: I tried to keep it as it was with the previous caseCreationReceiver and I believe the name suggests what it is doing hence did not change it.

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java (hunk 5 above)
Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:99]: yes my bad. will make the change.

Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java]: Actually it is in a way as one is logging a message when creating the subscription object and other is saying that now I am all set and listening to the pubsub. what you say?

File: _infra/helm/case/templates/deployment.yaml (hunk 1 above)
Comment (insacuri) [reply _infra/helm/case/templates/deployment.yaml:130]: so I did.  Sorry about that ^^

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java (hunk 2 above)
Comment (insacuri) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseCreationReceiver.java:37]: That's fine 👍

File: src/main/java/uk/gov/ons/ctp/response/casesvc/message/PubSub.java [hunk 6]
```diff
@@ -54,7 +54,7 @@ public ProjectSubscriptionName getCaseReceiptSubscriptionName() {
     String project = appConfig.getGcp().getProject();
     String subscriptionId = appConfig.getGcp().getReceiptSubscription();
     log.info(
-        "creating pubsub subscription name for case notification "
+        "creating pubsub subscription name for case creation notification "
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/message/PubSub.java:57]: Remove extra space after the word 'notification'

**Review by insacuri — APPROVED** [review]
2 very minor comments around spaces in logging, if you can fix them then great.  Otherwise you can just merge it

**Pull request merged**